From 71765f7c44090717d04ead082068e5acd49e40d4 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Thu, 12 Oct 2023 18:02:52 +0200
Subject: [PATCH 0673/1141] ASoC: stm32: sai: add pm runtime support in sai
 core

Enable pm_runtime support in SAI core.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: I9ae8ac27f9d1a99a890811f251b4cb089282f53e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/336783
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 sound/soc/stm/stm32_sai.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/sound/soc/stm/stm32_sai.c
+++ b/sound/soc/stm/stm32_sai.c
@@ -12,6 +12,7 @@
 #include <linux/module.h>
 #include <linux/of_platform.h>
 #include <linux/pinctrl/consumer.h>
+#include <linux/pm_runtime.h>
 #include <linux/reset.h>
 
 #include <sound/dmaengine_pcm.h>
@@ -267,6 +268,8 @@ static int stm32_sai_probe(struct platfo
 	sai->set_sync = &stm32_sai_set_sync;
 	platform_set_drvdata(pdev, sai);
 
+	pm_runtime_enable(&pdev->dev);
+
 	return devm_of_platform_populate(&pdev->dev);
 }
 
